packages/angular/cli/src/commands/generate/cli.ts

```diff
@@ -162,4 +160,45 @@ export class GenerateCommandModule
 
     return `${commandName}${positionalArgs ? ' ' + positionalArgs : ''}`;
   }
+
+  private async getSchematicsToRegister(): Promise<
+    [schematicName: string, collectionName: string][]
+  > {
+    const [, schematicNameFromArgs] = this.parseSchematicInfo(
+      // positional = [generate, component] or [generate]
+      this.context.args.positional[1],
+    );
+
+    /**
+     * - *key*: schematic name
+     * - *value*: collection name
+     */
+    const schematicsToRegister = new Map<string, string>();
+    const collectionNames = await this.getCollectionNames();
+
+    for (const collectionName of collectionNames) {
+      const workflow = this.getOrCreateWorkflowForBuilder(collectionName);
+      const collection = workflow.engine.createCollection(collectionName);
+      const schematicsInCollection = collection.description.schematics;
+
+      // We cannot use `collection.listSchematicNames()` as this doesn't return hidden schematics.
+      for (const schematicName of Object.keys(schematicsInCollection)) {
+        if (schematicsInCollection[schematicName].private) {
+          continue;
+        }
+
+        if (schematicNameFromArgs === schematicName) {
+          // No need to process all schematics since we know which one the user invoked.
+          return [[schematicName, collectionName]];
+        }
+
+        // If a schematic with this same name is already registered skip.
+        if (!schematicsToRegister.has(schematicName)) {
+          schematicsToRegister.set(schematicName, collectionName);
+        }
+      }
+    }
+
+    return [...schematicsToRegister.entries()].sort();
```

Comment:
`.entries()` is an array of key value pairs (arrays) right? How does `.sort()` compare each of these pairs? A quick look at [the MDN page](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/sort#description) says that `.sort()` will stringify it's inputs and compare the strings, which seems like it could lead to bugs. An explicit compare function seems like a good idea here.

Reply:
Haha, I tend to prefer the spread :) But okay!

I do agree about the sort.